app/src/main/java/eu/kanade/presentation/more/settings/screen/SettingsDataScreen.kt

```diff
@@ -111,7 +112,12 @@ object SettingsDataScreen : SearchableSettings {
                 val flags = Intent.FLAG_GRANT_READ_URI_PERMISSION or
                     Intent.FLAG_GRANT_WRITE_URI_PERMISSION
 
-                context.contentResolver.takePersistableUriPermission(uri, flags)
+                // For some reason InkBook devices do not implement the SAF properly. Persistable URI grants do not
+                // work. However, simply retrieving the URI and using it works fine for these devices. Access is not
+                // revoked after the app is closed or the device is restarted.
+                if (!isInkBookEInkDevice()) {
+                    context.contentResolver.takePersistableUriPermission(uri, flags)
+                }
```

Comment:
IMO wrap this in try-catch and show a notification on failure is better than a hardcoded blacklist.

Reply:
Sure, I can also do that. I'll use a toast notification.